fastly/block_fastly_service_v1_logging_ftp.go

```diff
@@ -0,0 +1,268 @@
+			},
+
+			"format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Apache-style string or VCL variables to use for log formatting.",
+			},
+
+			"format_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either 1 or 2. (default: 2)",
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+
+			"response_condition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the condition to apply",
+			},
+		},
+	},
+}
+
+func processFTP(d *schema.ResourceData, conn *fst.Client, latestVersion int) error {
+	serviceID := d.Id()
+	of, nf := d.GetChange("logging_ftp")
+
+	if of == nil {
+		of = new(schema.Set)
+	}
+	if nf == nil {
+		nf = new(schema.Set)
+	}
+
+	ofs := of.(*schema.Set)
+	nfs := nf.(*schema.Set)
+
+	removeFTPLogging := ofs.Difference(nfs).List()
+	addFTPLogging := nfs.Difference(ofs).List()
+
+	// DELETE old FTP logging endpoints
+	for _, oRaw := range removeFTPLogging {
+		of := oRaw.(map[string]interface{})
+		opts := buildDeleteFTP(of, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly FTP logging endpoint removal opts: %#v", opts)
+
+		if err := deleteFTP(conn, opts); err != nil {
+			return err
+		}
+	}
+
+	// POST new/updated FTP logging endpoints
+	for _, nRaw := range addFTPLogging {
+		ef := nRaw.(map[string]interface{})
+		opts := buildCreateFTP(ef, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly FTP logging addition opts: %#v", opts)
+
+		if err := createFTP(conn, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readFTP(conn *fst.Client, d *schema.ResourceData, s *fst.ServiceDetail) error {
+	// refresh FTP
+	log.Printf("[DEBUG] Refreshing FTP logging endpoints for (%s)", d.Id())
+	ftpList, err := conn.ListFTPs(&fst.ListFTPsInput{
+		Service: d.Id(),
+		Version: s.ActiveVersion.Number,
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERR] Error looking up FTP logging endpoints for (%s), version (%v): %s", d.Id(), s.ActiveVersion.Number, err)
+	}
+
+	ell := flattenFTP(ftpList)
+
+	if err := d.Set("logging_ftp", ell); err != nil {
+		log.Printf("[WARN] Error setting FTP logging endpoints for (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func createFTP(conn *fst.Client, i *fst.CreateFTPInput) error {
+	_, err := conn.CreateFTP(i)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteFTP(conn *fst.Client, i *fst.DeleteFTPInput) error {
+	err := conn.DeleteFTP(i)
+
+	if errRes, ok := err.(*fst.HTTPError); ok {
+		if errRes.StatusCode != 404 {
+			return err
+		}
+	} else if err != nil {
+		return err
+	}
```

Comment:
Wouldn't a 404 error returned from `DeleteFTP` result in a nil error being returned from this function? Is that expected?

Reply:
Why can't we simply have the following here:

```golang
return conn.DeleteFTP(i)
```